app/src/main/java/de/tum/in/tumcampusapp/component/tumui/grades/GradesActivity.kt

```diff
@@ -0,0 +1,394 @@
+package de.tum.`in`.tumcampusapp.component.tumui.grades
+
+import android.animation.Animator
+import android.animation.AnimatorListenerAdapter
+import android.animation.LayoutTransition
+import android.graphics.drawable.Animatable
+import android.os.Bundle
+import android.util.ArrayMap
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.widget.AdapterView
+import android.widget.AdapterView.OnItemSelectedListener
+import android.widget.ArrayAdapter
+import com.github.mikephil.charting.data.*
+import de.tum.`in`.tumcampusapp.R
+import de.tum.`in`.tumcampusapp.api.tumonline.CacheControl
+import de.tum.`in`.tumcampusapp.component.other.generic.activity.ActivityForAccessingTumOnline
+import de.tum.`in`.tumcampusapp.component.tumui.grades.model.Exam
+import de.tum.`in`.tumcampusapp.component.tumui.grades.model.ExamList
+import kotlinx.android.synthetic.main.activity_grades.*
+import java.text.NumberFormat
+import java.util.*
+
+/**
+ * Activity to show the user's grades/exams passed.
+ */
+class GradesActivity : ActivityForAccessingTumOnline<ExamList>(R.layout.activity_grades) {
+
+    // exams data and list
+    private var exams: List<Exam>? = null
```

Comment:
`exams` is passed around as argument several times and assigned but never read as member variable.